server/src/main/scala/org/http4s/server/middleware/VirtualHost.scala

```diff
@@ -29,10 +30,10 @@ object VirtualHost {
   def exact[F[_]](
       service: HttpService[F],
       requestHost: String,
-      port: Option[Int] = None): HostService[F] =
+      port: Option[Uri.Port] = None): HostService[F] =
```

Comment:
I don't think we can deprecate old versions because after erasure it's the same signature.